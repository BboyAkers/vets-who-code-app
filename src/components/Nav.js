import React, { useState, useRef, useEffect } from 'react'
import Link from 'gatsby-link'
import FluidImage from '../components/FluidImage'

function Nav() {
  const navRef = useRef()
  const [isNavOpen, setIsNavOpen] = useState(false)
  const [opacity, setOpacity] = useState(0.9)

  useEffect(() => {
    let current = true
    if (current) {
      document.addEventListener('scroll', handleScroll)
    }
    return () => {
      current = false
      document.removeEventListener('scroll', handleScroll)
    }
  })

  function handleScroll() {
    const winScroll = document.body.scrollTop || document.documentElement.scrollTop
    if (winScroll === 0) {
      setOpacity(0.9)
    }
    if (winScroll > 0) {
      setOpacity(1)
    }
  }

  return (
    <nav
      ref={navRef}
      id="fixedTopNav"
      className="navbar navbar-fixed-top main-navigation navbar-solid"
      itemScope=""
      itemType="https://schema.org/SiteNavigationElement"
      role="navigation"
      style={{ opacity: opacity }}
    >
      <div className="container">
        <div className="navbar-header">
          <div className="navbar-brand" itemScope="" itemType="https://schema.org/Organization">
            <span className="sr-only">#VetsWhoCode</span>
            <Link to="/">
              <FluidImage
                fileName="hashflag_white.jpg"
                alt="#VetsWhoCode Logo"
                className="logo_holder"
              />
            </Link>
          </div>
          <button
            type="button"
            id="hamburger-1"
            onClick={() => setIsNavOpen(!isNavOpen)}
            className={`navbar-toggle collapsed hamburger ${isNavOpen ? 'is-active' : ''}`}
            aria-expanded={isNavOpen ? 'true' : 'false'}
          >
            <span className="line"></span>
            <span className="line"></span>
            <span className="line"></span>
            <span className="sr-only">#VetsWhoCode</span>
          </button>
        </div>
        <div
          className={`navbar-collapse collapse ${isNavOpen ? 'in' : ''}`}
          id="main-nav-collapse"
          aria-expanded={isNavOpen ? 'true' : 'false'}
        >
          <ul className="nav navbar-nav navbar-right" role="menu">
            <li role="menu-item">
              {' '}
              <Link to="/">
                <span>Home</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/about">
                <span>About</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/syllabus">
                <span>Syllabus</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/mentor">
                <span>Mentor</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/apply">
                <span>Apply</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/donate">
                <span>Donate</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/contact">
                <span>Contact Us</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <a
                href="https://medium.com/vets-who-code"
                without="true"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>Blog</span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  )
}

export default Nav
