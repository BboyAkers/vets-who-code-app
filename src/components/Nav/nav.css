/* overide bootstrap mobile nav styles */

/* hamburger menu styles */
.hamburger {
  position: absolute;
  top: 15px;
  right: 8px;
}

.hamburger .line {
  width: 39px;
  height: 2px;
  background-color: #ecf0f1;
  display: block;
  margin: 8px auto;
  transition: all 0.3s ease-in-out;
}

.hamburger:hover {
  cursor: pointer;
}

.hamburger:focus {
  border: 1px dashed #fff;
}

#hamburger-1.is-active .line:nth-child(2) {
  opacity: 0;
}

#hamburger-1.is-active .line:nth-child(1) {
  transform: translateY(10px) rotate(45deg);
}

#hamburger-1.is-active .line:nth-child(3) {
  transform: translateY(-10px) rotate(-45deg);
}

.navbar-brand a {
  min-width: 200px;
  display: grid;
  grid-template-columns: 50px 180px;
  grid-template-rows: 50px;
}

.navbar-brand a .logo_holder {
  grid-column: 1/1;
  grid-row: 1/1;
}

.navbar-brand a .homeLink {
  grid-column: 2/2;
  grid-row: 1/1;
  padding-left: 5px;
  padding-top: 15px;
  vertical-align: middle;
  font-size: 0.5em;
}
@media only screen and (max-width: 363px) {
  .navbar-brand {
    max-width: 200px;
  }
}

/*  */
.donate:hover {
  transition: 0.3s linear;
  background-color: #c5203e;
}

@media (max-width: 1140px) {
  .navbar-right {
    float: right !important;
    margin-right: -15px;
    display: none;
  }
  .main-navigation .navbar-toggle {
    display: block;
    position: absolute;
    float: right;
    padding: 9px 10px;
    margin-top: 8px;
    margin-right: 15px;
    margin-bottom: 8px;
    background-color: transparent;
    background-image: none;
    border: 1px solid transparent;
  }
}
