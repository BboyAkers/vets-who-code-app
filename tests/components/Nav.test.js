import React from 'react'
import Nav from '../../src/components/Nav'

describe('<Nav />', () => {
  let wrapper

  beforeEach(() => wrapper = shallow(<Nav />))

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
