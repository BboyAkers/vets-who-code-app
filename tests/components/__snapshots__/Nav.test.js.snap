// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render correctly 1`] = `
<nav
  className="navbar navbar-fixed-top main-navigation"
  id="fixedTopNav"
  itemScope=""
  itemType="https://schema.org/SiteNavigationElement"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <button
        aria-expanded="false"
        className="navbar-toggle collapsed"
        data-target="#main-nav-collapse"
        data-toggle="collapse"
        type="button"
      >
         
        <span
          className="sr-only"
        >
          #VetsWhoCode
        </span>
         
        <span
          className="ion-drag"
        />
      </button>
      <div
        className="navbar-brand"
        itemScope=""
        itemType="https://schema.org/Organization"
      >
         
        <span
          className="sr-only"
          itemProp="name"
        >
          #VetsWhoCode
        </span>
        <GatsbyLink
          itemProp="url"
          to="/"
        >
          <img
            alt="#VetsWhoCode Logo"
            className="logo_holder"
            src="image-mock"
          />
        </GatsbyLink>
      </div>
    </div>
    <div
      aria-expanded="false"
      className="navbar-collapse collapse"
      id="main-nav-collapse"
      style={
        Object {
          "height": "1px",
        }
      }
    >
      <ul
        className="nav navbar-nav navbar-right"
        role="menu"
      >
        <li>
           
          <GatsbyLink
            to="/"
          >
            <span>
              Home
            </span>
          </GatsbyLink>
        </li>
        <li>
           
          <GatsbyLink
            to="/about"
          >
            <span>
              About
            </span>
          </GatsbyLink>
        </li>
        <li>
           
          <GatsbyLink
            to="/syllabus"
          >
            <span>
              Syllabus
            </span>
          </GatsbyLink>
        </li>
        <li>
           
          <GatsbyLink
            to="/mentor"
          >
            <span>
              Mentor
            </span>
          </GatsbyLink>
        </li>
        <li>
           
          <GatsbyLink
            to="/apply"
          >
            <span>
              Apply
            </span>
          </GatsbyLink>
        </li>
        <li>
           
          <GatsbyLink
            to="/donate"
          >
            <span>
              Donate
            </span>
          </GatsbyLink>
        </li>
        <li>
           
          <GatsbyLink
            to="/contact"
          >
            <span>
              Contact Us
            </span>
          </GatsbyLink>
        </li>
        <li>
           
          <a
            href="https://medium.com/vets-who-code"
            target="_blank"
          >
            <span>
              Blog
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
